src/plugins/discord/mirror.js

```diff
@@ -0,0 +1,384 @@
+// @flow
+
+import type Database from "better-sqlite3";
+import stringify from "json-stable-stringify";
+import * as Model from "./models";
+import dedent from "../../util/dedent";
+
+const VERSION = "DISCORD_MIRROR_v1";
+
+/**
+ * Persists a local copy of data from a Discord Guild.
+ * Implements create and read functionality.
+ *
+ * Each Mirror instance is tied to a particular Guild. Trying to use a mirror
+ * for multiple Discord Guilds is not permitted; use separate Mirrors.
+ */
+
+export class Mirror {
+  +_db: Database;
+
+  /**
+   * Construct a new Mirror instance.
+   *
+   * Takes a Database and GuildId.
+   *
+   */
+
+  constructor(db: Database, guildId: Model.Snowflake) {
+    if (db == null) throw new Error("db: " + String(db));
```

Comment:
Updated

Reply:
This pattern is actually used in both the other Plugins, but not clear to me why that would be.

Reply:
In some cases I think we've been "extra defensive" in checking for runtime nulls where flow already prohibits. Speaking for myself, I think it's a habit from working on untyped codebases... I think since we are using flow, they are unnecessary and can be safely skipped.